assets/js/state/sagas/softwareUpdatesSettings.test.js

```diff
@@ -44,4 +51,134 @@ describe('Software Updates Settings saga', () => {
       });
     });
   });
+
+  describe('Saving Software Updates settings', () => {
+    it('should successfully save software updates settings', async () => {
+      const axiosMock = new MockAdapter(networkClient);
+      const payload = {
+        url: faker.internet.url(),
+        username: faker.word.noun(),
+        password: faker.word.noun(),
+        ca_cert: faker.lorem.text(),
+      };
+      const caUploadedAt = faker.date.recent().toString();
+      const successfulResponse = softwareUpdatesSettingsFactory.build({
+        url: payload.url,
+        username: payload.username,
+        ca_uploaded_at: caUploadedAt,
+      });
+
+      axiosMock
+        .onPost('/settings/suma_credentials')
+        .reply(201, successfulResponse);
+
+      const dispatched = await recordSaga(saveSoftwareUpdatesSettings, payload);
+
+      expect(dispatched).toEqual([
+        startLoadingSoftwareUpdatesSettings(),
+        setSoftwareUpdatesSettings(successfulResponse),
+      ]);
+    });
+
+    it('should have errors on failed saving', async () => {
+      const axiosMock = new MockAdapter(networkClient);
+      const payload = {
+        url: '',
+        username: '',
+        password: faker.word.noun(),
+        ca_cert: '',
+      };
+      const errors = [
```

Comment:
OCD alert: mismatching errors based on the input. 
Providing an empty username `username: '',` would result in an extra error besides url/ca_cert.

Feel free to keep like this, tho :smile:  The test still tests what it's supposed to test.